cli/src/commands/push.ts

```diff
@@ -40,19 +64,31 @@ export default class Push extends Command {
       if (flags.context !== undefined) {
         throw new Error('--context flag should only be used with the --dockerfile flag')
       }
+      if (flags['include-git-commit']) {
+        throw new Error(
+          'The --include-git-commit flag can only be used when building a fresh image. Please provide a Dockerfile with the --dockerfile flag.',
+        )
+      }
       const { os, arch } = getImagePlatform(args.image)
       if (os !== 'linux' || arch !== 'amd64') {
         this.log()
-        this.warn(chalk.bold.red`The image ${args.image} may not be compatible with Jamsocket because its image os/arch is ${os}/${arch}.`)
-        this.log('If you encounter errors while spawning with this image, you may need to rebuild the image with the platform flag (--platform=linux/amd64) and push again.')
+        this.warn(
+          chalk.bold
+            .red`The image ${args.image} may not be compatible with Jamsocket because its image os/arch is ${os}/${arch}.`,
+        )
+        this.log(
+          'If you encounter errors while spawning with this image, you may need to rebuild the image with the platform flag (--platform=linux/amd64) and push again.',
+        )
         this.log()
 
-        const response = await inquirer.prompt([{
-          name: 'goAhead',
-          message: lightMagenta('Go ahead and push image?'),
-          type: 'list',
-          choices: [{ name: 'no' }, { name: 'yes' }],
-        }])
+        const response = await inquirer.prompt([
+          {
+            name: 'goAhead',
+            message: lightMagenta('Go ahead and push image?'),
+            type: 'list',
+            choices: [{ name: 'no' }, { name: 'yes' }],
+          },
+        ])
```

Comment:
I know this is just a formatting change, but since my eyes are on it, I wonder if a better approach here is to say that it is not compatible (instead of "may not be compatible"), and require people to pass an `--ignore-arch-check` (or `--i-understand-the-consequences-ignore-arch-check-anyway`) flag.

Thinking about how someone (Ethan?) encountered this warning, but thought that since he was prompted to skip it it wasn't a big deal. I can't really think of _any_ case where someone would want to ignore it unless they were e.g. a BYOC customer who ran their own aarch64 drones, but nobody has asked for that yet.

Reply:
(for a separate PR, of course)